plugins/dell-dock/fu-dell-dock-common.c

```diff
@@ -43,3 +43,19 @@ fu_dell_dock_set_power(FuDevice *device, guint8 target, gboolean enabled, GError
 
 	return fu_dell_dock_ec_modify_lock(parent, target, enabled, error);
 }
+
+const gchar *
+fu_dell_dock_get_instance_id(guint8 *type, DOCK_HWDEV_INFO *dev_list, guint16 *vid, guint16 *pid)
+{
+	for (guint i = 0; i < sizeof(dev_list); i++) {
+		if (*type != dev_list[i].dock_type)
+			continue;
+		if (dev_list[i].vid != 0 && vid != NULL && dev_list[i].vid != *vid)
+			continue;
+		if (dev_list[i].pid != 0 && pid != NULL && dev_list[i].pid != *pid)
+			continue;
+		return g_strdup(dev_list[i].instance_id);
```

Comment:
AFAICT the instance ID string is always a hardcoded string so this should be a memory leak.  You should just be returning the string.

Reply:
fixed in 271ccab.